src/unmanaged.rs

```diff
@@ -372,7 +372,10 @@ where
     #[inline]
     fn take(&mut self, key: K) -> V {
         let default = self.default.clone();
-        mem::replace(&mut self[key], default)
+        match self.try_get_mut(key) {
+            Some(val) => mem::replace(val, default),
```

Comment:
Might be worth a comment that this'll leave trailing `default`s - if those are at the end of the `data` vector, they could be trimmed to save memory, but the only resizing methods truncate (i.e. happily drop non-default values), which makes this hard (albeit not impossible) for the owner/client/caller to do.